### PR TITLE
github actions workflow to run tox

### PR DESCRIPTION
**Pull request by thraxil**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -14,7 +14,12 @@ deps =
 commands =
     py.test {posargs}
     coverage xml
-    codecov
-    pylint edx_sga
+#    codecov
+#    pylint edx_sga
 
 
+[gh-actions]
```
Comment (OmarIthawi) [inline tox.ini:21]: I don't trust this plugin. Tox is great, its plugins aren't always easy to work with on local development environments.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    branches: [master]
+  pull_request:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.5]
```
Comment (OmarIthawi) [inline .github/workflows/tests.yml:14]: Can we use the same pattern from this comment: https://github.com/appsembler/edx-organizations/pull/22#pullrequestreview-904530722

**Review by OmarIthawi — COMMENTED** [review]
I'm not sure what's the problem with the test failures. But I'm duplicating my suggestion to avoid using `tox-gh-actions` in favor of native `tox` configs.

**Comment by OmarIthawi** [comment]
@thraxil tests are failing due to unpinned dependencies:

https://github.com/appsembler/edx-sga/blob/378a0177276576637877f7e86107ddc12ed6401c/test_requirements.txt#L24-L27

So `xblock-sdk` needs something much lower than "latest".

The wisita XBlock has a bit more sane versions for a start:

 - https://github.com/appsembler/xblock-wistia/blob/master/requirements-test.txt

**Comment by thraxil** [comment]
looks like it's failing on py35 because it's pulling in a version of `xblock` that has 3.6+ syntax:

```
py35-django2 run-test: commands[0] | py.test
Traceback (most recent call last):
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/bin/py.test", line 8, in <module>
    sys.exit(console_main())
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/_pytest/config/__init__.py", line 187, in console_main
    code = main()
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/_pytest/config/__init__.py", line 143, in main
    config = _prepareconfig(args, plugins)
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/_pytest/config/__init__.py", line 319, in _prepareconfig
    pluginmanager=pluginmanager, args=args
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/_pytest/helpconfig.py", line 100, in pytest_cmdline_parse
    config = outcome.get_result()  # type: Config
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/_pytest/config/__init__.py", line 1003, in pytest_cmdline_parse
    self.parse(args)
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/_pytest/config/__init__.py", line 1280, in parse
    self._preparse(args, addopts=addopts)
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/_pytest/config/__init__.py", line 1187, in _preparse
    early_config=self, args=args, parser=self._parser
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/pluggy/manager.py", line 87, in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/pytest_django/plugin.py", line 353, in pytest_load_initial_conftests
    _setup_django()
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/pytest_django/plugin.py", line 236, in _setup_django
    django.setup()
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/django/apps/registry.py", line 114, in populate
    app_config.import_models()
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/django/apps/config.py", line 211, in import_models
    self.models_module = import_module(models_module_name)
  File "/opt/hostedtoolcache/Python/3.5.10/x64/lib/python3.5/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 985, in _gcd_import
  File "<frozen importlib._bootstrap>", line 968, in _find_and_load
  File "<frozen importlib._bootstrap>", line 957, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 697, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/workbench/models.py", line 12, in <module>
    from xblock.fields import BlockScope, Scope
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/xblock/__init__.py", line 11, in <module>
    import xblock.core
  File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/xblock/core.py", line 103
    raise DisallowedFileError(f"Only files from {cls.public_dir!r}/ are allowed: {uri!r}")

SyntaxError: invalid syntax
```

**Comment by thraxil** [comment]
Similar issue, but this time with `web_fragments`: https://github.com/openedx/web-fragments which isn't directly installed via `test_requirements.txt`, so some other package must be pulling it in. So we need to figure out what version we can pin that at that will still work on Python 3.5. Their CHANGELOG just says "check the git commit history" :(

**Comment by thraxil** [comment]
Again, 3.5 incompatible code being pulled in:

```
edx_sga/tests/test_showanswer.py:15: in <module>
    from workbench.runtime import WorkbenchRuntime
E     File "/home/runner/work/edx-sga/edx-sga/.tox/py35-django2/lib/python3.5/site-packages/workbench/runtime.py", line 137
E       usage_id = f"{def_id}.u{next(id_seq)}"
```
This appears to be coming from `xblock-sdk`, which is specified in `requirements.txt` with `git+https://github.com/edx/xblock-sdk.git#egg=xblock-sdk`

So, again, an unpinned dependency that we'll have to track down an appropriate version of...

**Comment by shadinaif** [comment]
@OmarIthawi I've pinned test packages to those used in edx-platform. The only change for `modified_time` affects only test files. No worries about it

**Review by shadinaif — APPROVED** [review]

**Pull request merged**